### PR TITLE
[HunyuanImage][Perf] opt prepare_attention_mask for e2e latency 6% reduction

### PR DESCRIPTION
**Pull request by Bounty-hunter**
**Before submitting:** run the precheck-pr skill with code agent for a self-check against project conventions — catches dead code, missing benchmarks, and title format issues.

PLEASE FILL IN THE PR DESCRIPTION HERE ENSURING ALL CHECKLIST ITEMS (AT THE BOTTOM) HAVE BEEN CONSIDERED.

## Purpose
For prepare attention mask
### Before:  (1) construct in cpu -> (2) cp to device
image

### After: just construct in device
image

## Test Plan
(1) run perf bench
` pytest -s -v tests/dfx/perf/scripts/run_diffusion_benchmark.py   --test-config-file tests/dfx/perf/tests/test_hunyuan_image_tp4.json`

(2) run accuracy test
`pytest -s -v tests/e2e/accuracy/test_hunyuan_image3_pixel_accuracy.py -m full_model --run-level full_model`

## Test Result
(1)
| case | tps | e2e latency | improve |
| -----| -----| ----| ----|
| Before | 0.2115 | 4.7287 s | - |
| After | 0.2248 | 4.4482 s | 6% |

Our expected improve is 2~3%, but we test with 6% improve; the reason is that:
The execution time of the original prepare_attention_mask function on the CPU exhibits substantial variability, ranging from 6 ms to 447 ms in our tests. While our change keep it less than 1ms.

(2)
image

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan. Please provide the test scripts & test commands. Please state the reasons if your codes don't require additional test scripts. For test file guidelines, please check the [test style doc](https://docs.vllm.ai/projects/vllm-omni/en/latest/contributing/ci/tests_style/)
- [ ] The test results. Please paste the results comparison before and after, or the e2e results.
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model. **Please run `mkdocs serve` to sync the documentation editions to `./docs`.**
- [ ] (Optional) Release notes update. If your change is user-facing, please update the release notes draft.
</details>

**BEFORE SUBMITTING,** please read <https://github.com/vllm-project/vllm-omni/blob/main/CONTRIBUTING.md>
(anything written below this line will be removed by GitHub Actions)

### CODE REVIEW

**Comment by Bounty-hunter** [comment]
@princepride @gcanlin @hsliuustc0106 PTAL

**Comment by hsliuustc0106** [comment]
is there any other similar situation?

**Review by hsliuustc0106 — APPROVED** [review]
lgtm

**Comment by Bounty-hunter** [comment]
> is there any other similar situation?

Yes, continue to identify and address them in subsequent PRs.
